game-core/src/config.rs

```diff
@@ -10,24 +11,51 @@ pub struct Ally {
 impl Default for Ally {
     fn default() -> Self {
         Ally {
+            max_hp: 0,
             follow_distance: 0.0,
             max_distance: 0.0,
             min_distance: 0.0,
         }
     }
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Player {
+    pub max_hp: u32,
+}
+
+impl Default for Player {
```

Comment:
just use add Default to the derive list rather than implementing one that defaults everything to zeros, I don't see particular value in having default values coded into the source. I'm okay with defaulting things to Zero but my preference would be to just not have these types implement default and error out if we forget to add a field to the config file.

If you want to have default impls I'd prefer the default and the override both exist as config files, in which case I'd recommend using https://github.com/mehcode/config-rs for loading the config files. Though I'm very surprised to see that it doesn't list ron as a supported config file format. /shrug